### PR TITLE
Fixed defmt on direct-boot mode

### PR DESCRIPTION
**Pull request by TheButlah**
Closes: #175 

Ideally, I would also like to add this for the regular boot mode. I have not tested that yet.

Author is @9names, I merely submitted the PR.

### CODE REVIEW

File: esp32c3-hal/ld/db-riscv-link.x [hunk 1]
```diff
@@ -110,6 +110,16 @@ SECTIONS
...
 
+  /* ### .uninit */
+  .uninit (NOLOAD) : ALIGN(4)
+  {
+    . = ALIGN(4);
+    __suninit = .;
+    *(.uninit .uninit.*);
+    . = ALIGN(4);
+    __euninit = .;
+  } > REGION_BSS
```
Comment (bjoernQ) [inline esp32c3-hal/ld/db-riscv-link.x:121]: nitpicking but maybe we should define an alias in `db-esp32c3-memory.x` like it's done for other regions - doesn't change anything but would look more consistent

**Comment by bjoernQ** [comment]
I'd be fine with merging this since it solves the original problem
Ideally all the linker scripts should be a bit more aligned but that is not just about this new section so definitely not the scope of this PR

@jessebraham fine with this?

**Comment by jessebraham** [comment]
No complaints from me!

File: esp32c3-hal/ld/db-riscv-link.x (hunk 1 above)
Comment (TheButlah) [reply esp32c3-hal/ld/db-riscv-link.x:121]: I merely reposted the original solution from @9names, I have no experience with linker scripts so I'm not sure how to do that 😅

Feel free to modify the pr though!

**Review by jessebraham — APPROVED** [review]

**Pull request merged**